File: LinkedList/doubleLL.cpp
#include<iostream>
using namespace std;

class Node{
  public:
    int data;
    Node* next;
    Node* prev;  
};

int main(){
    Node* head;
    Node* one=NULL;
    Node* two=NULL;
    Node* three=NULL;

    one=new Node();
    two=new Node();
    three=new Node();
    
    one->data=10;
    two->data=20;
    three->data=30;

    one->next=two;
    two->next=three;
    three->next=NULL;

    one->prev=NULL;
    two->prev=one;
    three->prev=two;

    head=one;
    int dir=0; //dir=1 for moving backwords and 0 for moving forward.
    while(head!=NULL){
        cout<<head->data<<endl;
        if(head->next==NULL){
            dir=1;
        }
        if(!dir){
            head=head->next;
        }
        else{
            head=head->prev;
        }

    }
}

File: LinkedList/doubleLL.js
class Node{
    constructor(item){
         this.data=item;
         this.next=null;
         this.prev=null;
    }
}

class LinkedList{
    constructor(){
        this.head=null;
    }
}

let ll=new LinkedList()

ll.head=new Node(15)
let two=new Node(25)
let three=new Node(35)

ll.head.next=two
ll.head.prev=null
two.next=three
two.prev=ll.head
three.prev=two
three.next=null

let dir=0

while(ll.head!=null){
   console.log(ll.head.data)
   if(ll.head.next==null){
       dir=1
   }
   if(dir==0){
       ll.head=ll.head.next
   }
   else{
       ll.head=ll.head.prev
   }
}